iib/workers/tasks/greenwave.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import json
+import logging
+
+import requests
+
+from iib.exceptions import IIBError
+from iib.workers.config import get_worker_config
+from iib.workers.tasks.utils import get_image_labels
+
+log = logging.getLogger(__name__)
+
+
+def gate_bundles(bundles, greenwave_payload):
+    """
+    Check if all bundle images have passed gating tests in the CVP pipeline.
+
+    This function queries Greenwave to check if the policies are satisfied for each bundle image.
+
+    :param list bundles: a list of strings representing the pull specifications of the bundles to
+        be gated.
+    :param dict greenwave_payload: the dict of payload required to query Greenwave to gate bundles.
+    :raises IIBError: if any of the bundles fail the gating checks or IIB fails to get a
+        response from Greenwave.
+    """
+    conf = get_worker_config()
+    _validate_greenwave_params_and_config(conf, greenwave_payload)
+
+    log.info('Gating on bundles: %s', ', '.join(bundles))
+    gating_unsatisfied_bundles = []
+    for bundle in bundles:
+        koji_build_nvr = _get_koji_build_nvr(bundle)
+        log.info('Querying Greenwave for decision on %s', koji_build_nvr)
+        greenwave_payload['subject_identifier'] = koji_build_nvr
```

Comment:
In my opinion, it's a bad practice to modify function parameters unless the function makes it obvious since the caller likely does not expect it.